File: spring-boot/src/main/java/com/example/demo/model/Pokemon.java
package com.example.demo.model;

import com.example.demo.dto.EvolutionChainDTO;
import com.example.demo.dto.PokemonDTO;
import com.example.demo.enums.Type;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Pokemon {

    @JsonProperty("id")
    private long id;

    @JsonProperty("name")
    private String name;

    @Enumerated(EnumType.STRING)
    @JsonProperty("type_one")
    private Type typeOne;

    @Enumerated(EnumType.STRING)
    @JsonProperty("type_two")
    private Type typeTwo;

    @JsonProperty("base_hp")
    private int baseHp;

    @JsonProperty("base_atk")
    private int baseAtk;

    @JsonProperty("base_def")
    private int baseDef;

    @JsonProperty("base_spa")
    private int baseSpa;

    @JsonProperty("base_spd")
    private int baseSpd;

    @JsonProperty("base_spe")
    private int baseSpe;

    @JsonProperty("base_total")
    private int baseTotal;

    @JsonProperty("sprite")
    private String sprite;

    @JsonProperty("icon_b64")
    private String icon_b64;

    @JsonProperty("stage")
    private int stage;

    public Pokemon(PokemonDTO pokemonDTO) {
        this.id = pokemonDTO.getId();
        this.name = pokemonDTO.getName();
        this.typeOne = pokemonDTO.getTypeOne();
        this.typeTwo = pokemonDTO.getTypeTwo();
        this.baseHp = pokemonDTO.getBaseHp();
        this.baseAtk = pokemonDTO.getBaseAtk();
        this.baseDef = pokemonDTO.getBaseDef();
        this.baseSpa = pokemonDTO.getBaseSpa();
        this.baseSpd = pokemonDTO.getBaseSpd();
        this.baseSpe = pokemonDTO.getBaseSpe();
        this.baseTotal = pokemonDTO.getBaseTotal();
        this.sprite = pokemonDTO.getSprite();
        this.icon_b64 = pokemonDTO.getIcon_b64();
        this.stage = pokemonDTO.getStage();
    }
}

File: angular/src/app/model/pokemon.ts
import { Type } from '../enum/type';

export class Pokemon {
    id: number;
    name: string;
    sprite: string;
    type_one: Type;
    type_two: Type | null;
    base_hp: number;
    base_atk: number;
    base_def: number;
    base_spa: number;
    base_spd: number;
    base_spe: number;
    base_total: number;
    icon_b64: string;
    stage: number;

    constructor(
        id: number,
        name: string,
        sprite: string,
        type_one: Type,
        type_two: Type | null,
        base_hp: number,
        base_atk: number,
        base_def: number,
        base_spa: number,
        base_spd: number,
        base_spe: number,
        base_total: number,
        icon_b64: string,
        stage: number
    ) {
        this.id = id;
        this.name = name;
        this.sprite = sprite;
        this.type_one = type_one;
        this.type_two = type_two;
        this.base_hp = base_hp;
        this.base_atk = base_atk;
        this.base_def = base_def;
        this.base_spa = base_spa;
        this.base_spd = base_spd;
        this.base_spe = base_spe;
        this.base_total = base_total;
        this.icon_b64 = icon_b64;
        this.stage = stage;
    }
}
